src/tests/integration-test/src/api_robustness_test.cpp

```diff
@@ -315,6 +315,25 @@ BOOST_AUTO_TEST_CASE(TestSQLFetchPastEnd) {
   SQLCloseCursor(stmt);
 }
 
+BOOST_AUTO_TEST_CASE(TestSQLFetchEmpty) {
```

Comment:
This test case seems unrelated to the fix which is about SQLGetData

Reply:
Previously this would set SQL_NO_TOTAL, which is not valid.

Reply:
This test is getting the return value for SQLFetch though, not SQLGetData.

Reply:
Wups, I thought I saw the test failing. I will isolate SQLGetData in another test.

Reply:
I did not see your SQLGetData test.

Reply:
Added the test for `SQLGetData` with an empty response.